common/locale.go

```diff
@@ -0,0 +1,47 @@
+package common
+
+import (
+	"net/http"
+	"strings"
+)
+
+//LangEN is a reference to the english short-form localeCode "en"
+const LangEN = "en"
+
+//LangCY is a reference to the english short-form localeCode "cy"
+const LangCY = "cy"
+
+// SetLocaleCode sets the Locale code used to set the language
+func SetLocaleCode(req *http.Request) *http.Request {
+	localeCode := ExtractLangFromSubDomain(req)
+
+	// Language is overridden by cookie 'lang' here if present.
+	if c, err := req.Cookie("lang"); err == nil && len(c.Value) > 0 {
+		localeCode = ExtractLangFromCookie(c)
+	}
+	req.Header.Set("LocaleCode", localeCode)
+
+	return req
+}
+
+// ExtractLangFromSubDomain returns a language based on subdomain
+func ExtractLangFromSubDomain(req *http.Request) string {
+	args := strings.Split(req.Host, ".")
+	if len(args) == 0 {
+		// Defaulting to "en" (LangEN) if no arguments
+		return LangEN
+	}
+	if strings.Split(req.Host, ".")[0] == LangCY {
+		return LangCY
+	}
+	return LangEN
+}
+
+// ExtractLangFromCookie returns a language based on the lang cookie
+func ExtractLangFromCookie(c *http.Cookie) string {
```

Comment:
🤔I think this function name is not 100% representative of the functionality. With my über picky hat on 🤠 I think it should make clear to the caller that it will use a default if its not an accepted value. I would either:
 - Rename it to something like I think something like `GetLangCookieOrDefault` OR
 - Return an empty string if its not `en` or `cy`  and have a separate `DefaultIfBlank(lang string)` func which does the check.

Sorry if this seems like a really picky comment. But I believe that a method should do what it says and no more - its then very clear to the caller what they can expect it to do.